### PR TITLE
 custom ArrayList and HushSet

### PR DESCRIPTION
**Pull request by MaVaYuran**

### CODE REVIEW

File: src/main/java/MyArrayList.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if (size == array.length) {
+            resizeArray();
+        }
+        array[size++] = element;
+    }
+
+    public T getByIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
```
Comment (Alexmansar) [inline src/main/java/MyArrayList.java]: could be one extra method. You use it twice

File: src/main/java/MyArrayList.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+public class MyArrayList<T> {
+    private T[] array;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyArrayList() {
+        this.array = (T[]) new Object[10];
```
Comment (Alexmansar) [inline src/main/java/MyArrayList.java]: 10 could be a constant. As ypu add in HashSet

File: src/main/java/MyHashSet.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            int index = getIndex(element);
+            data[index] = element;
+            size++;
+        }
+    }
+
+    private boolean contains(T element) {
+        int index = getIndex(element);
+        return data[index] != null && data[index].equals(element);
+    }
```
Comment (Alexmansar) [inline src/main/java/MyHashSet.java]: move after pubic methods

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@MaVaYuran Good job)) just few comments)

File: src/main/java/MyArrayList.java (hunk 1 above)
Comment (MaVaYuran) [reply src/main/java/MyArrayList.java]: you mean extract  method?

Comment (Alexmansar) [reply src/main/java/MyArrayList.java]: yes)

**Review by Alexmansar — APPROVED** [review]
@MaVaYuran could be merged)

**Pull request merged**